packages/core/src/fhir-deduplication/deduplicate-fhir.ts

```diff
@@ -355,13 +361,28 @@ function removeDanglingReferences<T extends Resource>(entry: T, link: string): T
       if (!entry.participant.length) delete entry.participant;
     }
   }
-  if ("performer" in entry) {
-    if (entry.resourceType === "DiagnosticReport") {
-      entry.performer = entry.performer?.filter(performer => performer.reference !== link);
-      if (!entry.performer.length) delete entry.performer;
-    } else if (entry.resourceType === "Procedure") {
```

Comment:
the issue is that we were only removing practitioners for DiagnosticReport and Procedure resources.

Reply:
Its possible this issue is elsewhere as well though testing on the consolidated bundle did not show it for other resources

Reply:
`performer` might not be an Array, as well as the code that filters the references is complicated and not very type safe (relies on converting to `any`).

I was checking the structure of the objects that have `performer` as prop, and ended up rewriting this, what do you think?

```ts
  if ("performer" in entry) {
    if (Array.isArray(entry.performer)) {
      if (
        entry.resourceType === "DiagnosticReport" ||
        entry.resourceType === "Observation" ||
        entry.resourceType === "ServiceRequest"
      ) {
        entry.performer = entry.performer.filter(p => p.reference !== link);
      } else if (
        entry.resourceType === "Immunization" ||
        entry.resourceType === "MedicationAdministration" ||
        entry.resourceType === "MedicationDispense" ||
        entry.resourceType === "MedicationRequest" ||
        entry.resourceType === "Procedure" ||
        entry.resourceType === "RiskAssessment"
      ) {
        entry.performer = entry.performer?.filter(p => p.actor?.reference !== link);
      }
      if (entry.performer.length === 0) {
        delete entry.performer;
      }
    } else {
      if (entry.performer.reference === link) {
        delete entry.performer;
      }
    }
  }
```

Can find the permutations of types by cmd+click (or F12) on `entry.performer`.

Reply:
I dont think we use any of the resources where Performer isnt an array but its good to have the protection here since we have no type safety or relationship to the handlebars code which we can change and then break things